### PR TITLE
Read multiple Datastores concurrently for queries

### PR DESCRIPTION
**Pull request by dvaldivia**
This PR introduces an `mpsc` to allow reading from all datastores in a log concurrently

### CODE REVIEW

File: src/query.rs [hunk 1]
```diff
@@ -249,79 +251,88 @@ impl Query {
...
-                                                        .1;
-
-                                                    evaluate_query_on_line(&q, &q_parse, line)
-                                                })
-                                                .map_err(|e| {
+                            let query_state_holder3 = Arc::clone(&query_state_holder);
+
+                            let (tx, rx) = mpsc::unbounded_channel();
+
+                            for i in 0..logs_ds_len {
```
Comment (donatello) [inline src/query.rs:260]: Can we also do some code improvement in this PR? For example each task per data store can be in its own function.

Comment (dvaldivia) [reply src/query.rs:260]: ready

File: src/storage.rs [hunk 2]
```diff
@@ -299,6 +299,7 @@ pub fn read_file_line_by_line(
                 // max line length of 1MiB
                 LinesCodec::new_with_max_length(1024 * 1024),
```
Comment (donatello) [inline src/storage.rs:300]: Not using the LineTaker yet?

Comment (dvaldivia) [reply src/storage.rs:300]: it is used as we discussed, also renamed to `TakeFromIterable`

**Review by donatello — APPROVED** [review]

**Pull request merged**